src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResultSimulationResource.java

```diff
@@ -0,0 +1,98 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.participation.*;
+import de.tum.in.www1.artemis.service.*;
+import de.tum.in.www1.artemis.service.ProgrammingSubmissionResultSimulationService;
+
+@Profile("dev")
+@RestController
+@RequestMapping("/api")
+public class ProgrammingSubmissionResultSimulationResource {
+
+    private final Logger log = LoggerFactory.getLogger(ProgrammingSubmissionResource.class);
+
+    private final ProgrammingSubmissionService programmingSubmissionService;
+
+    private final UserService userService;
+
+    private final ParticipationService participationService;
+
+    private final WebsocketMessagingService messagingService;
+
+    private final ProgrammingExerciseService programmingExerciseService;
+
+    private final ProgrammingSubmissionResultSimulationService programmingSubmissionResultSimulationService;
+
+    public ProgrammingSubmissionResultSimulationResource(ProgrammingSubmissionService programmingSubmissionService, UserService userService,
+            ParticipationService participationService, WebsocketMessagingService messagingService, ProgrammingExerciseService programmingExerciseService,
+            ProgrammingSubmissionResultSimulationService programmingSubmissionResultSimulationService) {
+        this.programmingSubmissionService = programmingSubmissionService;
+        this.userService = userService;
+        this.participationService = participationService;
+        this.messagingService = messagingService;
+        this.programmingExerciseService = programmingExerciseService;
+        this.programmingSubmissionResultSimulationService = programmingSubmissionResultSimulationService;
+    }
+
+    /**
+     * This method is used to create a participation and a submission
+     * This participation and submission are only SIMULATIONS for the testing
+     * of programming exercises without local setup
+     *
+     * @param exerciseID the id of the exercise
+     * @return HTTP OK and ProgrammingSubmission
+     */
+
+    @PostMapping(value = "submissions/no-local-setup/{exerciseID}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ProgrammingSubmission> createParticipationAndSubmissionSimulation(@PathVariable Long exerciseID) {
+
+        ProgrammingSubmission programmingSubmission = programmingSubmissionResultSimulationService.createSubmission(exerciseID);
+
+        programmingSubmissionService.notifyUserAboutSubmission(programmingSubmission);
+
+        return ResponseEntity.ok().body(programmingSubmission);
```

Comment:
Should return `201 Created` instead of `200 OK`.

Reply:
i changed it to 200